python_training/sim_city.py

```diff
@@ -0,0 +1,183 @@
+BASE_MUNICIPALITY_TAX_RATE = 1000
+
+
+class Ackland:
+    """
+    The state Ackland itself.
+    """
+    def __init__(self, *cities):
+        """
+        Initiates the Ackland state object.
+        """
+        self.cities = list(cities)
+
+    def calculate_total_tax_amount(self)->int:
+        """
+        Calculates the amount of tax money that would be earned from all the cities of Ackland.
+        :return: The total amount of tax money from all the cities of the state.
+        """
+        return sum([city.how_much_money() for city in self.cities])
+
+
+class City:
+    """
+    A city in Ackland.
+    """
+    def __init__(self, city_name, *neighborhoods):
+        """
+        Initiates a new City object.
+        :param neighborhoods: A list of Neighborhood objects that are part of the city.
+        """
+        # Check that there are no neighborhoods with the same name (and will be overwritten)
+        if len(set(neighborhood.name for neighborhood in neighborhoods)) != len(neighborhoods):
+            print("You have entered one or more neighborhoods with the same name.")
+        self.city_name = city_name
+        self.base_municipality_tax_rate = BASE_MUNICIPALITY_TAX_RATE
+        self.neighborhoods = {neighborhood.name: neighborhood for neighborhood in neighborhoods}
+
+    def how_much_money(self)->int:
+        """
+        Calculates the concluded amount of tax money that would be payed by The cities in Ackland.
+        :return: The amount of money that should be payed.
+        """
+        return self.base_municipality_tax_rate + sum([neighborhood.get_neighborhood_tax_rate() for neighborhood
+                                                     in self.neighborhoods.values()])
+
+    def ruin_a_neighborhood(self, neighborhood_name):
+        """
+        Removes a neighborhood from the city.
+        :param neighborhood_name: The name of the neighborhood that would be removed.
+        """
+        if neighborhood_name in self.neighborhoods:
+            self.base_municipality_tax_rate *= 1.05
+            del(self.neighborhoods[neighborhood_name])
```

Comment:
Don't use `del`, since it can have unexpected side-effects.
Better to use `pop` for these cases.